### PR TITLE
[Student][Tests][MBL-13720]: A batch of fixes for intermittently failing tests.

### PR DESCRIPTION
**Pull request by joehoag**
Mostly focussed on Discussion and Inbox interaction tests.

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

File: automation/canvas_espresso/src/main/kotlin/com/instructure/canvas/espresso/CanvasTest.kt [hunk 1]
```diff
@@ -77,7 +77,7 @@ abstract class CanvasTest : InstructureTest(BuildConfig.GLOBAL_DITTO_MODE) {
                 eventObject.put("status", "retry")
                 eventObject.put("testName", testMethod)
                 eventObject.put("testClass", testClass)
-                eventObject.put("stackTrace", error.stackTrace.take(10).joinToString(", "))
+                eventObject.put("stackTrace", error.stackTrace.take(15).joinToString(", "))
```
Comment (joehoag) [inline automation/canvas_espresso/src/main/kotlin/com/instructure/canvas/espresso/CanvasTest.kt]: The 10-deep stack trace wasn't cutting it for some failures, so I'm increasing it to 15.

File: automation/canvas_espresso/src/main/kotlin/com/instructure/canvas/espresso/mockCanvas/endpoints/CourseEndpoints.kt [hunk 2]
```diff
@@ -593,6 +593,7 @@ object CourseDiscussionEntryEndpoint : Endpoint(
                 val jsonObject = grabJsonFromMultiPartBody(request.body()!!)
                 Log.d("<--", "topic entry replies post body: $jsonObject")
                 val newEntry = DiscussionEntry(
+                        id = data.newItemId(),
```
Comment (joehoag) [inline automation/canvas_espresso/src/main/kotlin/com/instructure/canvas/espresso/mockCanvas/endpoints/CourseEndpoints.kt]: Not really a stability fix, but (1) it's the right thing to do, and (2) it helped me get some ad-hoc multi-reply tests to run.

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1015210&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Pull request merged**